ls1intum/Artemis
# Enhancement/programming exercise/import remove test trigger

### Checklist

### Motivation and Context


This should resolve #951. We have to remove the build triggers from the test repository to the template build plans when importing (old) programming exercises. Additionally, we only want to allow the import of generated exercises, meaning linked exercises should not be made available.

### Description

I modified the SQL queries for searching for importable exercise. They now check, if the short name is not null, which is only the case for generated exercises.

There is also an additional parameter for the copy of build plans allowing to specify any number of repository names that should trigger the plan. If this list is not null, all triggers will get overwritten, otherwise the already existing triggers will get copied without touching them.

Edit: Either the new Bamboo CLI or the new Bamboo instance don't set the same permissions for copied projects, meaning that only administrators were able to see the copied build plans afterwards. I added a call to the Bamboo REST API setting the permissions like they were in the original project (read permission for logged in users. If you don't have this permission, even if you are allowed to see the plan, you were not able to do so) 

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Import an old programming exercise
4. The imported exercise template plan should not have a trigger to the test repository any more.
